### DISCUSSION TITLE
Circular Buffer

### DISCUSSION THREAD

**Discussion by homesenola** [mtconnect/cppagent#389] [Q&A]
Hii,

What is the size of circularBuffer in terms of bytes....?
I got to know that the circularBuffer can store 2^17 observations by default, does this mean it can store 2^17 observations?....... also what is the maximum size for each observations?

**Comment by wsobel** [comment]
I will need to compute the size of each entry. There are some entries in the circular buffer that take up more space since the "observations" are dynamically allocated. If they contain something like a data set, they can be fairly large.

For a "average" sample or event type, they should be fairly compact. It would be good to get the stats so we can advise people. I'll see if I can instrument the code to determine the size given various conditions.
